docs/2026-04-03-autoresearch-design.md

```
@@ -0,0 +1,745 @@
+2. If `currentPhase === "bootstrap"`: ensure branch, ensure results.tsv, run baseline if needed
+3. Build cycle prompt via `promptCompiler.buildCyclePrompt(state, recentResults)`
+4. Call `neurolink.generate()` with:
+   - `systemPrompt`: from `promptCompiler.buildSystemPrompt()`
+   - `tools`: from `createResearchTools(deps)`
+   - `maxSteps`: 20 (enough for propose→edit→commit→run→evaluate→record→accept/revert)
+   - `thinkingLevel`: from config
+   - `prepareStep`: from `createPrepareStepCallback(stateStore)`
+   - `provider` + `model`: from config
+5. After `generate()` returns, read state to determine outcome
+6. Apply deterministic keep/discard policy:
+   ```typescript
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)